addons/binding/org.openhab.binding.mihome.test/src/test/groovy/org/openhab/binding/mihome/discovery/test/MiHomeSubdeviceDiscoveryOSGiTest.groovy

```diff
@@ -0,0 +1,210 @@
+
+        inbox.approve(result.getThingUID(),result.getLabel())
+        waitForAssert{
+            MiHomeSubdevicesHandler handler = getThingHandler(MiHomeSubdevicesHandler)
+            assertThat "ThingHandler should be created after inbox approval for DiscoveryResult ${result}",handler,is(notNullValue())
+            ThingUID uid = handler.getThing().getUID()
+            assertThat "Thing UID and DiscoveryResult UID aren't the same",uid,is(equalTo(result.getThingUID()))
+            assertThat "Thing isn't intialized completely",handler.getThing().getStatus(),is(ThingStatus.ONLINE)
+            // This thing will be removed after the test
+            subdeviceThing =  handler.getThing()
+        }
+        discoveryService.stopScan()
+    }
+
+    @Test
+    public void 'dont create a result when no subdevice are found' () {
+
+        serverResponseBody = generateJsonDevicesListServerResponse("success")
+
+        discoveryService.startScan();
+
+        List<DiscoveryResult> results = inbox.getAll()
+        int expectedResults = 0
+        assertThat "Exaclty ${expectedResults} DiscoveryResults were expected",results.size(),is(expectedResults)
+
+        discoveryService.stopScan()
+    }
+
+    @Test
+    public void 'dont create result for subdevice with a Thing'() {
+        int testDeviceID = 51816
+        String testDeviceType = DeviceTypesConstants.OPEN_SENSOR_TYPE
+        ThingTypeUID expectedThingTypeUID = MiHomeBindingConstants.THING_TYPE_OPEN_SENSOR
+
+        // Register a thing
+        subdeviceThing = createThing(thingRegistry,gatewayThing,expectedThingTypeUID,testDeviceID)
+        thingRegistry.add(subdeviceThing)
+
+        waitForAssert{
+            MiHomeSubdevicesHandler handler = subdeviceThing.getHandler()
+            assertThat handler, is(notNullValue())
+            assertThat subdeviceThing.getStatus(),is(ThingStatus.ONLINE)
+        }
+
+        // Set up the backend response
+        JsonSubdevice subdevice = new JsonSubdevice(testDeviceID,TEST_GATEWAY_ID,testDeviceType)
+        serverResponseBody = generateJsonDevicesListServerResponse("success", subdevice)
+
+        discoveryService.startScan();
+
+        List<DiscoveryResult> results = inbox.getAll()
+        int expectedResults = 0
+        assertThat "Exaclty ${expectedResults} DiscoveryResults were expected",results.size(),is(expectedResults)
+
+        discoveryService.stopScan();
+    }
+
+    @After
+    public void tearDown(){
+
+        unregisterServlet(PATH_CREATE_GATEWAY)
+
+        removeBridge(thingRegistry,gatewayThing)
+        gatewayThing = null
+
+        removeThing(thingRegistry,gatewayThing)
+        subdeviceThing = null
+
+        waitForAssert {
+            discoveryService = getService(DiscoveryService, MiHomeSubdeviceDiscoveryService)
+            assertThat "DiscoveryService has to be unregistered after the BridgeHandler is removed.",discoveryService, is(nullValue())
```

Comment:
Thanks, but I couldn't agree that this is applicable to the cases where waitForAssert is used.

In this case I want to be sure that the service is unregistered before the other test case starts.